core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php

```diff
@@ -53,11 +53,12 @@ protected function createAuthProvider(ContainerBuilder $container, $id, $config,
     {
         $twoFactorProviderId = TwoFactorFactory::PROVIDER_ID_PREFIX.$id;
         $twoFactorFirewallConfigId = 'contao.security.two_factor_firewall_config.'.$id;
-        $twoFactorFirewallConfig = new Definition(TwoFactorFirewallConfig::class, [$config, $id, null]);
+        $twoFactorFirewallConfig = new Definition(TwoFactorFirewallConfig::class, [$config, $id, null, null]);
 
         $container
             ->setDefinition($twoFactorFirewallConfigId, $twoFactorFirewallConfig)
             ->replaceArgument(2, new Reference('security.http_utils'))
+            ->replaceArgument(3, new Reference('scheb_two_factor.security.request_data_reader'))
```

Comment:
(Not necessary to the author of this PR)

Is there a reason why this isn't a `ChildDefinition`? At least you're using one for the other service definitions further down, so why not here?

If you use a child definition and configure the service based on the service definition template, you don't need the two `replaceArgument` statements further down and we could have saved us all the hassle with the BC break now.